sigstore/verify/models.py

```diff
@@ -239,20 +258,32 @@ def from_bundle(
                 f"expected exactly one log entry, got {len(tlog_entries)}"
             )
         tlog_entry = tlog_entries[0]
+        inclusion_proof = tlog_entry.inclusion_proof
+
+        parsed_inclusion_proof: LogInclusionProof | None = None
+        if inclusion_proof and offline:
+            checkpoint = inclusion_proof.checkpoint
+
+            # If the bundle to be verified offline includes an inclusion proof,
+            # we verify it, which requires a checkpoint.
+            if not checkpoint.envelope:
+                raise InvalidMaterials("expected checkpoint in inclusion proof")
+
+            parsed_inclusion_proof = LogInclusionProof(
+                checkpoint=checkpoint.envelope,
+                hashes=[h.hex() for h in inclusion_proof.hashes],
+                log_index=inclusion_proof.log_index,
+                root_hash=inclusion_proof.root_hash.hex(),
+                tree_size=inclusion_proof.tree_size,
+            )
```

Comment:
```suggestion

        # NOTE: We only load the inclusion proof from the bundle if we know that
        # we're performing an offline verification, since that's the only case
        # in which we'll use it.
        inclusion_proof = tlog_entry.inclusion_proof
        parsed_inclusion_proof: LogInclusionProof | None = None
        if inclusion_proof and offline:
            checkpoint = inclusion_proof.checkpoint

            # If the bundle to be verified offline includes an inclusion proof,
            # we verify it, which requires a checkpoint.
            if not checkpoint.envelope:
                raise InvalidMaterials("expected checkpoint in inclusion proof")

            parsed_inclusion_proof = LogInclusionProof(
                checkpoint=checkpoint.envelope,
                hashes=[h.hex() for h in inclusion_proof.hashes],
                log_index=inclusion_proof.log_index,
                root_hash=inclusion_proof.root_hash.hex(),
                tree_size=inclusion_proof.tree_size,
            )
```